chapters/classes.tex

```diff
@@ -276,36 +276,127 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
+    (Note that certain uses of a parameter requires it to be a structural parameter, thereby forcing these simplifications to be made.)
+  \item
+    The declaration equation is given by a structural expression (\cref{structural-expressions}).
+  \end{itemize}
+  A structural parameter remains constant during transient analysis, with a value either determined during translation (similar to having prefix \lstinline!constant!) or by the initialization problem (similar to a \willbedefined{normal parameter}, see item below).
+  At which of these stages the value is determined is tool dependent.
+\item
+  % TODO: Add "normal parameter" to index.
+  A variable \lstinline!vc! declared with the \lstinline!parameter! prefix, is called a \firstuse{normal parameter} unless it is a structural parameter.
+  It remains constant during transient analysis, with a value determined by the initialization problem.
+\item
+  A \emph{discrete-time} variable \lstinline!vd! has a vanishing time derivative between events.
+  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere,
+  as the derivative is not even defined at the events, and it is not legal
+  to apply \lstinline!der! to discrete-time variables as they are not continuous. During transient analysis the variable
+  can only change its value at event
+  instants (see \cref{events-and-synchronization}).
+\item
+  A \emph{continuous-time} variable \lstinline!vn! may have a non-vanishing time
+  derivative (\lstinline!der(vn)<>0! possible) and may also
+  change its value discontinuously at any time during transient analysis
+  (see \cref{events-and-synchronization}). If there are any discontinuities the variable is
+  not differentiable.
+\end{itemize}
+
+\begin{nonnormative}
+With every normal parameter, there is at least one reason why it isn't a structural parameter.
+This information is useful to maintain in tools, as it allows generation of informative error messages when a violation of structural expression variability is detected.
+For example:
+\begin{lstlisting}[language=modelica]
+parameter Integer n =
+  if b then 1 else 2;        /* Normal parameter due to cycle with 'b'. */
+parameter Boolean b = n > 3; /* Normal parameter due to cycle with 'n'. */
+Real[n] x;                   /* Variability error: 'n' must be structural. */
+\end{lstlisting}
+Here, a good error message for the variability error can include the information that the reason for \lstinline!n! being a normal parameter is that it is in a cycle with \lstinline!b!.
+
+(Would structural and normal parameters have had different variability prefixes in their component declarations, it would have been possible to detect and report errors directly at the structural component declarations instead.)
```

Comment:
Actually not.  This part of the non-normative block wants help the user understand why parameters in Modelica can't be as easily described as the other component variabilities.

Reply:
I don't like really this sentence (line 404 in classes.tex). Maybe it's the use of "would" that makes it sound to me as if we wish it was in the specification but alas. It just does't sit right me. It's like we are putting into the specification what we wish the specification was instead of just making it so. That's why I thought it was a note to self that accidentally got into the change. Thinking of how to rephrase this sentence, it also occurs to me that we don't want to have such prefixes. Whether a parameter is structural or normal might change depending on a modification in a component or while extending. Having a fixed prefix would prevent that.

Reply:
The purpose of the comment is to help the reader understand why it needs to be as complicated as the example
illustrates.  Surely, there are pros and cons of having the same prefix for two different things, but I can try to formulate it in a less wishful way.  I'm just too tired to think of something right now.  Maybe tomorrow.

Reply:
(I am not in a position to prepare commits due to Easter holidays, so I'll just present an idea for now.)

How about something like this?
> The need to explain why a parameter is normal is the price we pay for not needing to make an explicit choice between structural and normal parameter using different variability prefixes in the component declarations.

Reply:
This conversation is very old and outdated in view of the last design meeting discussions.  Resolving.